src/pages/UserProfile.js

```diff
@@ -17,47 +28,185 @@ export default function UserProfile() {
+                  id="gender"
+                  type="text"
+                  name="gender"
+                  value={gender}
+                  onChange={(event) => setGender(event.target.value)}
+                />
+                <SaveButton
+                  onClick={() =>
+                    changeInformation(singleUser, 'gender', gender)
+                  }
+                >
+                  Speichern
+                </SaveButton>
+              </>
+            )}
             <TitleStyled>Die Rasse deines Hundes:</TitleStyled>
-            <DescriptionStyled>{singleUser.breed}</DescriptionStyled>
+            {on || <DescriptionStyled>{singleUser.breed}</DescriptionStyled>}
+            {on && (
+              <>
+                <InputStyled
+                  id="breed"
+                  type="text"
+                  name="breed"
+                  value={breed}
+                  onChange={(event) => setBreed(event.target.value)}
+                />
+                <SaveButton
+                  onClick={() => changeInformation(singleUser, 'breed', breed)}
+                >
+                  Speichern
+                </SaveButton>
+              </>
+            )}
             <TitleStyled>Über euch:</TitleStyled>
-            <DescriptionStyled>{singleUser.about}</DescriptionStyled>
+            {on || <DescriptionStyled>{singleUser.about}</DescriptionStyled>}
+            {on && (
+              <>
+                <LongInputStyled
+                  id="about"
+                  type="text"
+                  name="about"
+                  value={about}
+                  onChange={(event) => setAbout(event.target.value)}
+                />
+                <SaveButton
+                  onClick={() => changeInformation(singleUser, 'about', about)}
+                >
+                  Speichern
+                </SaveButton>
+              </>
+            )}
             <TitleStyled>Wonach ihr sucht:</TitleStyled>
-            <DescriptionStyled>{singleUser.about}</DescriptionStyled>
+            {on || <DescriptionStyled>{singleUser.search}</DescriptionStyled>}
+            {on && (
+              <>
+                <LongInputStyled
+                  id="search"
+                  type="text"
+                  name="search"
+                  value={search}
+                  onChange={(event) => setSearch(event.target.value)}
+                />
+
+                <SaveButton
+                  onClick={() =>
+                    changeInformation(singleUser, 'search', search)
+                  }
+                >
+                  Speichern
+                </SaveButton>
+              </>
+            )}
           </ProfileWrapper>
         </>
       )}
     </main>
   )
+  function changeInformation(singleUser, key, value) {
+    db.collection('users')
+      .doc(singleUser.id)
+      .update({ [key]: value })
+      .then(() => {})
+      .catch((err) =>
+        alert(
```

Comment:
den alert kannst du auch durch einen swal ersetzen, dann ist es einheitlicher :)

Reply:
guck mal in die doku von swal, die haben coole anpassungen für den alert

Reply:
Oh stimmt, daran hab ich gar nicht gedacht, danke!